draft-ietf-tls-ctls.md

```diff
@@ -354,15 +247,43 @@ group is listed by the code point name in {{RFC8446}}, Section 4.2.7.
 (e.g., ed25519). This implies a literal "signature_algorithms" extension
 consisting solely of this group.
 
-randomSize (integer):
+random (integer):
 : indicates that the ClientHello.Random and ServerHello.Random values
 are truncated to the given values. When the transcript is
 reconstructed, the Random is padded to the right with 0s and the
 anti-downgrade mechanism in {{RFC8446)}, Section 4.1.3 is disabled.
 IMPORTANT: Using short Random values can lead to potential
-attacks. When Random values are shorter than 8 bytes, PSK-only modes
-MUST NOT be used, and each side MUST use fresh DH ephemerals.
-The Random length MUST be less than or equal to 32 bytes.
+attacks. The Random length MUST be less than or equal to 32 bytes.
+When this element is omitted, the random values are created by hashing 
+the ephemeral public key via the hash algorithm indicated in 
+cipherSuite and truncated to 32 bytes. 
+
+mutualAuth (boolean):
+: indicates whether the client has to be prepared to transmit a 
+Certificate and a CertificateVerify message and whether the server 
+has to transmit a CertificateRequest message. 
```

Comment:
Isn't this actually "we are going to pretend CR was sent but it wasnt really"?

Reply:
Yes. Should we phrase it differently?

Reply:
I think we would need to lay some groundwork earlier. Do we have a general statement for when this kind of optimization would be possible?

Reply:
The entire issue here is whether we want to pre-define that the client has to authenticate itself to the server. If so, then with TLS 1.3 the server has to send the CertificateRequest message. If we want to obmit this message then this configuration option was supposed to indicate this functionality. 

As a consequence, the Client would send a Certificate + CertificateVerify message without receiving a CertificateRequest message first. 

Is the term "mutualAuth" better replaced by something like "skipCertificateRequest" or "implicitCertificateRequest"? Maybe.